src/Grassmann.jl

```diff
@@ -0,0 +1,275 @@
+    end
+end
+function check_tangent_vector(G::Grassmann{N,K,T},x,v; kwargs...) where {N,K,T}
+    t = check_manifold_point(G,x)
+    if (t != nothing)
+        return t
+    end
+    if (T <: Real) && !(eltype(v) <: Real)
+        return DomainError(eltype(v),
+            "The matrix $(v) is not a real-valued matrix, so it can not be a tangent vector to the Grassmann manifold of dimension ($(N),$(K)).")
+    end
+    if (T <: Complex) && !(eltype(v) <: Real) && !(eltype(v) <: Complex)
+        return DomainError(eltype(v),
+            "The matrix $(v) is neiter real- nor complex-valued matrix, so it can not bea tangent vector to the complex Grassmann manifold of dimension ($(N),$(K)).")
+    end
+    if any( size(v) != representation_size(G) )
+        return DomainError(size(v),
+            "The matrix $(v) is does not lie in the tangent space of $(x) on the Grassmann manifold of dimension ($(N),$(K)), since its dimensions are wrong.")
+    end
+    if !isapprox(x'*v + v'*x, zeros(K,K); kwargs...)
+        return DomainError(norm(x'*v + v'*x),
+            "The matrix $(v) is does not lie in the tangent space of $(x) on the Grassmann manifold of dimension ($(N),$(K)), since x'v + v'x is not the zero matrix.")
+    end
+end
+
+@doc doc"""
+    distance(M,x,y)
+
+computes the Riemannian distance on [`Grassmann`](@ref) manifold `M`$= \mathrm{Gr}(n,k)$.
+
+Let $USV = {\bar x}^\mathrm{T}y$ denote the SVD decomposition of
+${\bar x}^\mathrm{T}y$, where ${\bar\cdot}^{\mathrm{T}}$ denotes the complex
+conjugate transposed. Then the distance is given by
+````math
+d_{\mathrm{GR}(n,k)}(x,y) = \operatorname{norm}(\operatorname{Re}(b)).
+````
+where
+
+$b_{i}=\begin{cases} 0 & \text{if} \; S_i≧1 \\ \operatorname{acos}(S_i) & \, \text{if} \; S_i<1 \end{cases}.$
+"""
+function distance(M::Grassmann{N,K,T}, x, y) where {N,K,T}
+    if x ≈ y
+		return 0.
+  	else
+        a = svd(x'*y).S
+        a[a .> 1] .= 1
+		return sqrt(sum( (acos.(a)).^2 ))
+  	end
+end
+
+@doc doc"""
+    exp!(M, y, x, v)
+
+compute the exponential map on the [`Grassmann`](@ref) manifold
+`M`$= \mathrm{Gr}(n,k)$ starting in `x` with tangent vector (direction) `v`
+and store the result in `y`.
+Let $v = USV$ denote the SVD decomposition of $v$.
+Then the exponential map is written using
+````math
+z = x V\cos(S){\bar V}^\mathrm{T} + U\sin(S){\bar V}^\mathrm{T},
+````
+where ${\bar\cdot}^{\mathrm{T}}$ denotes the complex conjugate transposed.
+The cosine and sine are applied element wise to the diagonal entries of $S$.
+A final QR decomposition $z=QR$ is performed for numerical stability reasons,
+yielding the result as
+````math
+y = Q.
+````
+"""
+function exp!(M::Grassmann{N,K,T},y, x, v) where {N,K,T}
+    if norm(M,x,v) ≈ 0
+        return (y .= x)
+    end
+    d = svd(v)
+    z =  x * d.V * Diagonal(cos.(d.S)) * d.Vt + d.U * Diagonal(sin.(d.S)) * d.Vt
+    # reorthonormalize
+    y = copyto!(y, Array(qr(z).Q) )
```

Comment:
```suggestion
    copyto!(y, Array(qr(z).Q) )
```
We don't need the assignment.